index.ts

```diff
@@ -58,82 +58,126 @@ export interface ServiceDetails<S> {
   expose: MethodDetails<S>[];
   service: string;
   help?: string;
+  path?: string;
 }
 
 export interface Service {
   [methodName: string]: Method;
 }
 
-export function createRequestHandler<S extends Service>(
-  service: S,
-  serviceDetails: ServiceDetails<S>
-): RequestHandler {
-  const exposed = serviceDetails.expose;
-  const methods = exposed.map((m) => m.methodName);
-  const serviceName = serviceDetails.service;
-  const serviceHelp = serviceDetails.help || serviceName + " service";
-
-  const meta = {
-    serviceName,
-    multiArg: false,
-    help: serviceHelp,
-    interfaces: exposed.map((method: MethodDetails<S>) => {
-      if (typeof method === "string") {
-        throw new Error(
-          "Schema for expose has changed. Please refer to @loke/http-rpc documentation."
+class RegisteredService<S extends Service> {
+  constructor(private service: S, private serviceDetails: ServiceDetails<S>) {}
+
+  createRequestHandler(): RequestHandler {
```

Comment:
This is a library so we should be adding method descriptions to help devs use it

```suggestion
  /**
   * Creates an express HTTP handler that will process RPC requests for methods exposed by the service, as well as return metadata about the service.
   */
  createRequestHandler(): RequestHandler {
```

Reply:
Adding comments now